### ISSUE TITLE
Samples are not easy to consume due to extensive usage of parents. 

### ISSUE THREAD

**Issue by wtrocki** [java-operator-sdk/java-operator-sdk#257]
The idea of the samples should be that I should be able to grab folders outside repo and they should work. 
From a maintenance standpoint and verification that might be bad as repo possibly needs to depend on the compiled examples etc. but from the usage point of view there is

This would be something that simple CLI/Maven Artifacts could fix. 
Happy to build CLI for java operator if needed and add some capability to generate operator.

**Comment by psycho-ir** [comment]
Thank you @wtrocki for bringing this topic up, indeed using the samples should be simpler than what it is. 
I agree with building an initiator to setup the project. 
It could be used to address #28 and the current issue. 
I would suggest we make a new repo for it rather than a new module here to ensure it's developed in an isolated environment and its releases not being dependent on the SDK releases. 

What do you think @adam-sandor @csviri @metacosm?

**Comment by metacosm** [comment]
I'd argue that the samples should maybe be in a different repository. A generator to generate a skeleton operator would also help.

**Comment by adam-sandor** [comment]
Separate repos for each sample or single samples repo? I'd vote for single samples repo, so when the framework changes we can upgrade the samples with one PR.

**Comment by csviri** [comment]
The original idea was, when we started to extract the samples, that we have just a few simple samples, that supports only the development maybe some quick checking / testing, that sample works with the update. 
And other samples (which are more complex) should be extracted to a separate repo, I would say it's own. And there should be not too much of them.

**Comment by adam-sandor** [comment]
I think single repo for all (non-test) samples would be better. Easier to browse all the samples and easier to upgrade them on FW change.

**Comment by csviri** [comment]
That is true that is easier to upgrade them. So maybe the examples which are kind of showcases we should have in one repo. 
Question is what about the memcached, if the intention is to have it as a production ready operator, that should be a separate repo. Not sure if that is the intention.

**Comment by adam-sandor** [comment]
I don't think memcached will be a production ready operator and it should actually go into the repo in the operator-sdk with the other memcached samples.

**Comment by csviri** [comment]
@adam-sandor you mean a separate repo with the samples, or directly to the SDK?

**Comment by adam-sandor** [comment]
here https://github.com/operator-framework/operator-sdk-samples

**Comment by csviri** [comment]
Ok, now I'm completely lost how you mean this, let's discuss it maybe on next weekly.

**Comment by wtrocki** [comment]
Is weekly open for community. Would love to join as listener?

**Comment by psycho-ir** [comment]
+1 for a repo with samples and deprecate the tomcat-operator.

**Comment by adam-sandor** [comment]
@wtrocki please jump in our Discord channel (https://discord.gg/DacEhAy) you can find the invite as a pinned message in the general channel. You are very welcome to join the weeklys.

**Comment by metacosm** [comment]
@charlottemach you mentioned you were interested in taking this on during the last weekly call, is it still the case?

**Comment by csviri** [comment]
I will close this for now because of inactivity.
